File: python/problem762.py
class Solution(object):
    def countPrimeSetBits(self, L, R):
        """
        :type L: int
        :type R: int
        :rtype: int
        """
        count=0
        li=[2,3,5,7,11,13,17,19]
        for i in xrange(L,R+1):
            n=str(bin(i)[2:])
            m=n.count('1')
            if m in li:
                count+=1
        return count

File: cpp/problem762.cpp
#include <iostream>
#include <unordered_set>
using namespace std;

// Class: Solution by hash-determined primer number and loop-calculated set-bit number.
// Description:
//  - time complexity O(R-L).
//  - space complexity Theta(1).
// Creat: Yangyao, date 20181003
// References: none.
class Solution {
public:
    int countPrimeSetBits(int L, int R) {
        int count = 0, count_setbit, temp;
        for (int i = L; i <= R; ++i){
            count_setbit = 0; temp = i;
            do{
                if( temp & 1 ) ++count_setbit;
            }while( temp >>= 1 );
            if( primes_int32.find( count_setbit ) != primes_int32.end() ) ++count;
        }
        return count;
    }
private:
    const unordered_set<int> primes_int32 = {2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31};
};

// Func: Driver function to test solutions
int main(int argc, char const *argv[]){
    Solution sol;
    cout << sol.countPrimeSetBits(1,100) << endl;
    return 0;
}
